packages/mui-joy/src/Select/Select.tsx

```diff
@@ -571,16 +571,24 @@ const Select = React.forwardRef(function Select<OptionValue extends {}, Multiple
     [listboxProps.modifiers],
   );
 
+  let displayValue = placeholder;
+
+  if (Array.isArray(selectedOption)) {
+    if (selectedOption.length > 0) {
+      displayValue = renderValue(selectedOption);
+    }
+  } else if (selectedOption) {
+    displayValue = renderValue(selectedOption);
+  }
```

Comment:
```suggestion
  if (
    (Array.isArray(selectedOption) && selectedOption.length > 0) ||
    (!Array.isArray(selectedOption) && !!selectedOption)
  ) {
    displayValue = renderValue(selectedOption);
  }
```

Reply:
This is more readable for me, especially considering we do the same in the two blocks of code.